File: GUI/src/app/shared/models/chatRoom/chat-room.ts
import { v4 as uuidv4 } from 'uuid';

export class ChatRoom {
    public id: string;
    recipientId: string;
    senderId: string;
    createdDate: Date;

    constructor() {
        this.id = uuidv4();
        this.recipientId = uuidv4();
        this.senderId = uuidv4();
      }
}

File: CoreServices/src/main/java/com/example/entities/ChatRoom.java
package com.example.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.GenericGenerator;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "chat_room")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class ChatRoom {
    @Id
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
    @Column(columnDefinition = "BINARY(16)")
    private UUID id;
    private UUID senderId;
    private UUID recipientId;
    private LocalDateTime createdDate;

    @PrePersist
    protected void onCreate() {
        createdDate = LocalDateTime.now();
    }
}
